lib/wallet/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -791,18 +791,15 @@ applyOurTxToUTxO !slot !blockHeight !s !tx !u0 =
     -- NOTE 2: We do not have in practice the actual input amounts, yet we
     -- do make the assumption that if one input is ours, then all inputs are
     -- necessarily ours and therefore, known as part of our current UTxO.
-    actualFee direction = case (tx ^. #fee, direction) of
-        (Just x, Outgoing) ->
-            -- Shelley and beyond:
-            Just x
-        (Nothing, Outgoing) ->
-            -- Byron:
-            let totalOut = F.fold (TxOut.coin <$> outputs tx)
-                totalIn = TB.getCoin spent
-            in
-            Just $ distance totalIn totalOut
-        (_, Incoming) ->
-            Nothing
+    actualFee direction = fromMaybe byronFee (tx ^. #fee)
+      where
+        byronFee :: Coin
+        byronFee =
+            case direction of
+                Incoming -> distance totalOut totalIn
+                Outgoing -> distance totalIn totalOut
```

Comment:
Unfortunately, these fee computations for the Byron era are, strictly speaking, _wrong_ — the transaction may contain inputs which do not belong to the wallet — they do not take part in the `distance` calculation, but they still affect the fees.

For `Outgoing` transactions, the chance is high that the transaction contains only inputs that belong to the wallet, hence this computation works most of the time. However, for `Incoming` transactions, this computation is wrong more often than not.

Reply:
One other observation: the `Coin.distance` function used here is **_commutative_**: it computes the **_absolute distance_** between two `Coins` (natural numbers), which is in turn another `Coin` (natural number). So we have:
```hs
∀ x y. distance x y == distance y x
```
However, the code (as it stands) seems to make a point of **_commuting_** the order of the `totalOut` and `totalIn` arguments:
```hs
                Incoming -> distance totalOut totalIn
                Outgoing -> distance totalIn totalOut
```
This might give readers the impression that `distance` is somehow non-commutative, i.e., that the argument order is somehow important.

But the above code will have exactly the same behaviour as:
```hs
                Incoming -> distance totalIn totalOut
                Outgoing -> distance totalIn totalOut
```
Which means that this whole function (as it is written) is equivalent to just:
```hs
        byronFee :: Coin
        byronFee = distance totalIn totalOut
```